exporter/verifier/platforms.go

```diff
@@ -62,10 +63,25 @@ func CheckInvalidPlatforms[T comparable](ctx context.Context, res *result.Result
 
 	if len(reqMap) == 1 && len(ps.Platforms) == 1 {
 		pp := platforms.Normalize(ps.Platforms[0].Platform)
-		if _, ok := reqMap[platforms.Format(pp)]; !ok {
-			return []client.VertexWarning{{
-				Short: []byte(fmt.Sprintf("Requested platform %q does not match result platform %q", req.Platforms[0], platforms.Format(pp))),
-			}}, nil
+		if _, ok := reqMap[platforms.FormatAll(pp)]; !ok {
+			// The requested platform will often not have an OSVersion on it, but the
```

Comment:
This was a little janky.
When the platform string is not exactly what was requested (i.e. no OS version) then the returned result which has the OS Version will trigger a warning.